docs/clustering/downscaling.rst

```diff
@@ -0,0 +1,360 @@
+
+Exploring the Data
+==================
+
+Using the `Admin UI`_, shards view on the left:
+
+.. image:: shards-view.png
+
+You can see the three nodes, with each having a number of shards, like so:
+
+    +-------+---+---+---+---+---+---+
+    | Shard | 0 | 1 | 2 | 3 | 4 | 5 |
+    +=======+===+===+===+===+===+===+
+    |  n1   | . | . | . |   | . |   |
+    +-------+---+---+---+---+---+---+
+    |  n2   | . | . |   | . |   | . |
+    +-------+---+---+---+---+---+---+
+    |  n3   |   |   | . | . | . | . |
+    +-------+---+---+---+---+---+---+
+
+Thus in this cluster setup, one node can crash; yet the data in the cluster will still
+remain fully available because any two nodes have access to all the shards when they
+work together to fulfill query requests. A SQL table is a composite of shards (six in
+our case). When a query is executed, the planner will define steps for accessing all
+the shards of the table. By adding nodes to the cluster, or shards to the table, or both,
+the data is spread over more nodes, so that the computing is parallelized.
+
+Having a look at the setup for table *logs*:
+
+::
+
+  SHOW CREATE TABLE logs;
+
+Will return:
+
+::
+
+  CREATE TABLE IF NOT EXISTS "doc"."logs" (
+     "log_time" TIMESTAMP WITH TIME ZONE NOT NULL,
+     "client_ip" IP NOT NULL,
+     "request" TEXT NOT NULL,
+     "status_code" SMALLINT NOT NULL,
+     "object_size" BIGINT NOT NULL
+  )
+  CLUSTERED INTO 6 SHARDS
+  WITH (
+
+     number_of_replicas = '0-1',
+
+  )
+
+You have a default min number of replicas of zero, and a max of one for each
+of our six shards. A replica is simply a copy of a shard.
+
+
+.. _scaling-down-downscaling:
+
+Downscaling to a single node cluster
+====================================
+
+Scaling down to a single node cluster is the extreme example. In general
+downscaling is achieved by making sure the surviving nodes of the cluster have
+access to all the shards, even when the other nodes are missing.
+
+The first step is to ensure that the number of replicas matches the number of
+nodes, so that all nodes have access to all the shards:
+
+::
+
+  ALTER TABLE logs SET (number_of_replicas = '1-all');
+
+In the `Admin UI`_, you can follow the progress of replication.
+
+When replication is completed, you need to take down all the nodes in the cluster,
+as you are going to externally affect is state by means of the crate-node-tool_. You
+will first detach nodes [n2, n3], and then will bootstrap node [n1]. For convenience
+here is how each operation is invoked from the command line:
+
+::
+
+    ./crate/bin/crate-node detach-cluster -Cpath.home=$(pwd)/crate \
+                                          -Cpath.conf=$path_conf \
+                                          -Cpath.data=$path_data \
+                                          -Cpath.repo=$path_snapshots \
+                                          -Cnode.name=<NODE_NAME> \
+                                          -Ctransport.tcp.port=<PORT>
```

Comment:
These settings are not needed by the crate-node tool (and also defined inside the template, or not needed at all like the repo.path)